core/io/http_client.cpp

```diff
@@ -95,6 +95,9 @@ Error HTTPClient::connect_to_host(const String &p_host, int p_port, bool p_ssl,
 
 void HTTPClient::set_connection(const Ref<StreamPeer> &p_connection) {
 	ERR_FAIL_COND_MSG(p_connection.is_null(), "Connection is not a reference to a valid StreamPeer object.");
+	ERR_FAIL_COND_MSG(
+			ssl && !p_connection->is_class_ptr(StreamPeerSSL::get_class_ptr_static()),
+			"Connection is not a reference to a valid StreamPeerSSL object.");
```

Comment:
The added check is correct but that's a fairly uncommon way to verify the type of an object in Godot code. A more common approach would be: `Object::cast_to<StreamPeerSSL>(p_connection)` or
```
Ref<StreamPeerSSL> ssl_peer = p_connection;
if (ssl_peer.is_null()) ...
```

Reply:
Thanks for your comments @akien-mga!

What are your thoughts about creating a predicate `is_instance_of` (that could be used like `Object::is_instance_of<StreamPeerSSL>(p_connection)`) to check if an object is an instance of a certain class? I think it would better express the intent when we just want to know if the object is of that class or not, and have no need for the "casted" objected (which is the case here).

Reply:
Well `cast_to` being used to check if it's an instance is already used in hundreds of locations throughout the engine so changing that is far beyond the scope of this PR :)

Reply:
Definitely not the scope of this PR, but my idea wasn't doing that find & replace work here; I'd just create the predicate and it would be there available to use whenever it made sense. But alright, I'll just stick with `Object::cast_to`. Thanks!